android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/login/StrongPasswordGenerator.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.login
+
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.concept.storage.Login
+import mozilla.components.feature.prompts.concept.SimplePromptView
+import mozilla.components.feature.prompts.consumePromptFrom
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * The [StrongPasswordGenerator] displays a [SimplePromptView] for a site after
+ * receiving a [PromptRequest.SelectLoginPrompt] when a user clicks into a login field and we don't have
+ * any matching logins. The user can receive a suggestion for a strong password that can be used for filling
+ * in the password field.
+ *
+ * @property store The [BrowserStore] this feature should subscribe to.
+ * @property suggestStrongPasswordBar The view where the suggest strong password "prompt" will be inflated.
+ * @property sessionId This is the id of the session which requested the prompt.
+ */
+internal class StrongPasswordGenerator(
+    private val store: BrowserStore,
+    private val suggestStrongPasswordBar: SimplePromptView,
+    private var sessionId: String? = null,
+) : SimplePromptView.Listener {
+
+    init {
+        suggestStrongPasswordBar.listener = this
+    }
+
+    internal fun handleSuggestStrongPasswordRequest(
+        request: PromptRequest.SelectLoginPrompt,
+    ) {
+        request.generatedPassword?.let { suggestStrongPasswordBar.showPrompt(it) }
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    fun dismissCurrentSuggestStrongPassword(promptRequest: PromptRequest.SelectLoginPrompt? = null) {
+        try {
+            if (promptRequest != null) {
+                promptRequest.onDismiss()
+                sessionId?.let {
+                    store.dispatch(ContentAction.ConsumePromptRequestAction(it, promptRequest))
+                }
+                suggestStrongPasswordBar.hidePrompt()
+                return
+            }
+
+            store.consumePromptFrom<PromptRequest.SelectLoginPrompt>(sessionId) {
+                it.onDismiss()
+            }
+        } catch (e: RuntimeException) {
+            Logger.error("Can't dismiss this prompt", e)
+        }
+        suggestStrongPasswordBar.hidePrompt()
+    }
+
+    override fun onUseGeneratedPassword(generatedPassword: String) {
+        store.consumePromptFrom<PromptRequest.SelectLoginPrompt>(sessionId) {
+            val createdLoginEntryWithPassword = Login(
```

Comment:
Presumably the other fields will be updated in another ticket?

Reply:
[sure](https://bugzilla.mozilla.org/show_bug.cgi?id=1869575)